src/main/java/net/fabricmc/meta/web/ProfileHandler.java

```diff
@@ -129,9 +130,18 @@ private static JsonObject buildProfileJson(LoaderInfoV2 info, String side) {
 
 		profile.addProperty("mainClass", launcherMeta.get("mainClass").getAsJsonObject().get(side).getAsString());
 
-		// I believe this is required to stop the launcher from complaining
 		JsonObject arguments = new JsonObject();
+
+		// I believe this is required to stop the launcher from complaining
 		arguments.add("game", new JsonArray());
+
+		if (side.equals("client")) {
+			// add '-DFabricMakeNVidiaHappy= net.minecraft.client.main.Main ' for hybrid notebook solutions and similar assume it's MC
+			JsonArray jvmArgs = new JsonArray();
+			jvmArgs.add("-DFabricMakeNVidiaHappy= net.minecraft.client.main.Main ");
```

Comment:
I have confirmed this seems to fix discord not detecting minecraft. Im happy go with this but might not be a bad idea to rename nvidia to something more generic?